src/etos_client/lib/log_handler.py

```diff
@@ -75,16 +76,23 @@ def all_logs(self):
             for log in self._logs(finished):
                 yield log
 
-    def _get_path(self, _):
-        """Path to store log in.
-
-        :param log_name: Log name to check.
-        :type log_name: str
-        :return: Path where to store log.
-        :rtype: str
-        """
-        # TODO: Detect artifact logs for artifact_dir.
-        return self.report_dir
+    @property
+    def all_artifacts(self):
```

Comment:
Readability suffers with all the nested for-loops and embedded function calls. Not a deal breaker but just a reflection.

Reply:
Maybe. I experimented a bit on moving the code around and it became worse, in my opinion.